src/datastore/pg-store-v2.ts

```diff
@@ -269,4 +279,133 @@ export class PgStoreV2 extends BasePgStoreModule {
       return statusArray;
     });
   }
+
+  async getPoxCycles(args: PoxCyclePaginationQueryParams): Promise<DbPaginatedResult<DbPoxCycle>> {
+    return this.sqlTransaction(async sql => {
+      const limit = args.limit ?? PoxCycleLimitParamSchema.default;
+      const offset = args.offset ?? 0;
+      const results = await sql<(PoxCycleQueryResult & { total: number })[]>`
+        SELECT
+          cycle_number, block_height, index_block_hash, total_weight, total_signers,
+          total_stacked_amount, COUNT(*) OVER()::int AS total
+        FROM pox_cycles
+        WHERE canonical = TRUE
+        ORDER BY cycle_number DESC
+        OFFSET ${offset}
+        LIMIT ${limit}
+      `;
+      return {
+        limit,
+        offset,
+        results: results,
+        total: results[0].total,
+      };
+    });
+  }
+
+  async getPoxCycle(args: PoxCycleParams): Promise<DbPoxCycle | undefined> {
+    return this.sqlTransaction(async sql => {
+      const results = await sql<PoxCycleQueryResult[]>`
+        SELECT
+          cycle_number, block_height, index_block_hash, total_weight, total_signers,
+          total_stacked_amount
+        FROM pox_cycles
+        WHERE canonical = TRUE AND cycle_number = ${args.cycle_number}
+        LIMIT 1
+      `;
+      if (results.count > 0) return results[0];
+    });
+  }
+
+  async getPoxCycleSigners(
+    args: PoxCycleParams & PoxSignerPaginationQueryParams
+  ): Promise<DbPaginatedResult<DbPoxCycleSigner>> {
+    return this.sqlTransaction(async sql => {
+      const limit = args.limit ?? PoxSignerLimitParamSchema.default;
+      const offset = args.offset ?? 0;
+      const cycleCheck =
+        await sql`SELECT cycle_number FROM pox_cycles WHERE cycle_number = ${args.cycle_number} LIMIT 1`;
+      if (cycleCheck.count === 0)
+        throw new InvalidRequestError(`PoX cycle not found`, InvalidRequestErrorType.invalid_param);
+      const results = await sql<(DbPoxCycleSigner & { total: number })[]>`
+        SELECT
+          signing_key, weight, stacked_amount, weight_percent, stacked_amount_percent,
+          COUNT(*) OVER()::int AS total
+        FROM pox_sets
+        WHERE canonical = TRUE AND cycle_number = ${args.cycle_number}
+        ORDER BY weight DESC
```

Comment:
I think it should be common for multiple entries to have the same weight, so I think this needs to also sort by another column so the order is deterministic per page. Probably something like
```sql
ORDER BY weight DESC, stacked_amount DESC, signing_key
```